cookiecutter/main.py

```diff
@@ -96,24 +53,13 @@ def cookiecutter(
     # If no config file, sensible defaults from config.DEFAULT_CONFIG are used
     config_dict = get_user_config(config_file=config_file)
 
-    template = expand_abbreviations(template, config_dict)
-
-    if is_repo_url(template):
-        repo_dir = clone(
-            repo_url=template,
-            checkout=checkout,
-            clone_to_dir=config_dict['cookiecutters_dir'],
-            no_input=no_input
-        )
-    else:
-        # If it's a local repo, no need to clone or copy to your
-        # cookiecutters_dir
-        repo_dir = template
-
-    if not os.path.isdir(repo_dir):
-        raise RepositoryNotFound(
-            'The repository {0} could not be located.'.format(template)
-        )
+    repo_dir = determine_repo_dir(
+        template=template,
+        abbreviations=config_dict.get('abbreviations', {}),
```

Comment:
``config_dict`` needs to provide a default value by itself. If it does not currently we need to extend the config module.